### PR TITLE
twister: Add option for generating rom.json and ram.json

### PR DESCRIPTION
**Pull request by oyvindronningstad**
Generate the files via the 'footprint' target for each project.

### CODE REVIEW

File: scripts/pylib/twister/twisterlib/environment.py [hunk 1]
```diff
@@ -359,6 +359,9 @@ def add_parse_arguments(parser = None):
     parser.add_argument("--enable-size-report", action="store_true",
                         help="Enable expensive computation of RAM/ROM segment sizes.")
 
+    parser.add_argument("--create-rom-ram-report", action="store_true",
+                        help="Enable generate rom.json and ram.json for all projects.")
+
```
Comment (PerMac) [inline scripts/pylib/twister/twisterlib/environment.py:364]: I'd suggest a bit more explicit description. E.g.
```
"Add `--target footprint` argument to cmake build call, generating detailed ram/rom json reports for each build.
```

Comment (oyvindronningstad) [reply scripts/pylib/twister/twisterlib/environment.py]: Thanks, rewrote it now based on your suggestion, please take a look

**Review by PerMac — APPROVED** [review]

**Comment by oyvindronningstad** [comment]
CI is green now

File: scripts/pylib/twister/twisterlib/runner.py [hunk 2]
```diff
@@ -1059,6 +1059,8 @@ def run(self):
         sys.stdout.flush()
 
     def gather_metrics(self, instance: TestInstance):
+        if self.options.create_rom_ram_report:
+            self.run_build(['--build', self.build_dir, "--target", "footprint"])
```
Comment (nashif) [inline scripts/pylib/twister/twisterlib/runner.py:1063]: this is calling the build twice, why not just call the main build with the target instead, something along the lines of:

```patch
--- a/scripts/pylib/twister/twisterlib/runner.py
+++ b/scripts/pylib/twister/twisterlib/runner.py
@@ -1029,7 +1029,10 @@ class ProjectBuilder(FilterBuilder):
         return res

     def build(self):
-        res = self.run_build(['--build', self.build_dir])
+        if self.options.create_rom_ram_report:
+            res = self.run_build(['--build', self.build_dir, "--target", "footprint"])
+        else:
+            res = self.run_build(['--build', self.build_dir])
         return res


```

Comment (oyvindronningstad) [reply scripts/pylib/twister/twisterlib/runner.py:1063]: Thanks for the review. I'm not a Ninja expert, but I assume building the "footprint" target means the default target is not built, so it's not guaranteed to be the same thing I think. Calling build twice doesn't redo the whole build (I assume), whatever's already been built will be skipped when doing the footprint target.

**Review by nashif — APPROVED** [review]

**Pull request merged**